addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1905,7 +1905,9 @@ var BasicModel = AbstractModel.extend({
         var view = fieldInfo.views && fieldInfo.views[fieldInfo.mode];
         var def, rec;
         var defs = [];
-        list._changes = list._changes || [];
+        if (_.isEmpty(list._changes)){
```

Comment:
As `fieldName` is an x2many field, it's datapoint is a 'list' datapoint, so if set, `_changes` is an array, not an object. In your scenario it isn't the case, so it means that either we get here whereas we shouldn't (don't know why), or we beforehand wrongly set `_changes` to an object on that datapoint. In both cases, this isn't the right fix. I'll take it over.

Reply:
Thanks for taking it. I tried to find a reason why it's object here and only found that overriding happens in _applyOnChange:

https://github.com/odoo/odoo/blob/c0e082ee1fba7cecc711f724974d79ed65bab29b/addons/web/static/src/js/views/basic/basic_model.js#L1667

but I didn't investigate further